### ISSUE TITLE
Dependency hell since 0756e6f83f3dc6423f07b82812556b1d96422d20

### ISSUE THREAD

**Issue by LEDfan** [onddo/owncloud-cookbook#22]
Hi 

Can I ask you why commit 0756e6f83f3dc6423f07b82812556b1d96422d20 was made? I can't resolve the dependencies of this cookbook anymore. I'm using librarian-chef:
````
[Librarian] Ruby Version: 2.1.3
[Librarian] Ruby Platform: x86_64-linux-gnu
[Librarian] Rubygems Version: 2.2.2
[Librarian] Librarian Version: 0.1.2
[Librarian] Librarian Adapter: chef
[Librarian] Librarian Adapter Version: 0.0.4
[Librarian] Project: /home/tobia/Workspace/Chat/vagrant
[Librarian] Specfile: Cheffile
[Librarian] Lockfile: Cheffile.lock
[Librarian] Git: /usr/bin/git
[Librarian] Git Version: 2.1.4
[Librarian] Git Environment Variables:
[Librarian]   (empty)
[Librarian] Pre-Cached Sources:
[Librarian]   [:site, "https://supermarket.getchef.com/api/v1", {}]
[Librarian]   [:git, "https://github.com/opscode-cookbooks/database", {:ref=>"v2.1.8"}]
[Librarian]   [:git, "https://github.com/opscode-cookbooks/mysql_chef_gem", {:ref=>"v0.0.2"}]
[Librarian]   [:git, "https://github.com/LEDfan/chef-php-recipes", {}]
[Librarian] Post-Cached Sources:
[Librarian]   [:site, "https://supermarket.getchef.com/api/v1", {}]
[Librarian]   [:git, "https://github.com/opscode-cookbooks/database", {:ref=>"v2.1.8"}]
[Librarian]   [:git, "https://github.com/opscode-cookbooks/mysql_chef_gem", {:ref=>"v0.0.2"}]
[Librarian]   [:git, "https://github.com/LEDfan/chef-php-recipes", {}]
[Librarian]   [:git, "https://github.com/onddo/owncloud-cookbook", {:ref=>"master"}]
[Librarian]   [:git, "https://github.com/LEDfan/chef-php-recipes", {:ref=>"master"}]
[Librarian] Analyzing spec and lock:
[Librarian]   Removed:
[Librarian]   ExplicitRemoved:
[Librarian]   Added:
[Librarian]   NonMatchingAdded:
[Librarian]   Changed:
[Librarian]     owncloud
[Librarian]   DeepKeep:
[Librarian]     database
[Librarian]     git
[Librarian]     mysql_chef_gem
[Librarian]     php-recipes
[Librarian]     phpunit
[Librarian]   ShallowStrip:
[Librarian]     owncloud
[Librarian] Resolving owncloud (>= 0) <https://github.com/onddo/owncloud-cookbook#master>
[Librarian]   Checking manifests
[Librarian]     --- No output
[Librarian]     --- No output
[Librarian]     --- No output
[Librarian]     --- No output
[Librarian]     --> origin
[Librarian]     -->   origin/HEAD -> origin/master
[Librarian]     -->   origin/master
[Librarian]     --> 0756e6f83f3dc6423f07b82812556b1d96422d20
[Librarian]     --> 0756e6f83f3dc6423f07b82812556b1d96422d20
[Librarian]     Checking owncloud/0.5.0 <https://github.com/onddo/owncloud-cookbook#master>
[Librarian]       Conflict between openssl (~> 4.0.0) <(no source specified)> and openssl (~> 2.0) <(no source specified)>
[Librarian]       Backtracking from owncloud/0.5.0 <https://github.com/onddo/owncloud-cookbook#master>
[Librarian]   Failed to resolve owncloud (>= 0) <https://github.com/onddo/owncloud-cookbook#master>
Could not resolve the dependencies.
````

Thanks!

**Comment by zuazo** [comment]
The cookbook was broken due to some backwards incompatible changes in some community cookbooks. For example, the MySQL cookbook:

```ruby
Chef::Exceptions::RecipeNotFound
--------------------------------
could not find recipe mysql for cookbook database
```

We added version constraints to all the dependency cookbooks to avoid such problems in the future.

Surely we should update the openssl cookbook version in the metadata.

**Comment by zuazo** [comment]
I updated the openssl cookbook version in master. Does this solve your problem?

**Comment by LEDfan** [comment]
Yes that solves the issue! Know I don't have to specify any cookbooks version in my Cheffile anymore! Great.

**Comment by zuazo** [comment]
Fix released in [0.5.0](https://github.com/onddo/owncloud-cookbook/releases/tag/0.5.0).
